source/extensions/filters/network/zookeeper_proxy/filter.cc

```diff
@@ -293,6 +465,22 @@ void ZooKeeperFilter::onConnectResponse(const int32_t proto_version, const int32
                                         const std::chrono::milliseconds& latency) {
   config_->stats_.connect_resp_.inc();
 
+  if (!config_->latency_threshold_map_.empty()) {
```

Comment:
let's move those logic into ZookeeperFilterConfig by following steps
(1) move `default_latency_threshold_` to Config as it belongs to the proto
(2) introduce a enum class {NONE, SLOW, FAST}
(3) open an public function from ZookeeperFilterConfig like errorbudgetDecision and we can do
switch (config->errorbudgetDecision(OpCode::Connect, latency) {
  ...
}
(4) move `latency_threshold_map_` into private section


I know this is tedious but I though we can do better abstraction and DRY over connect and other opcode. wdyt?

Reply:
Thank you very much for your suggestions!
I think they are really good ideas, which make the code high cohesion and low coupling.